tests/specs/lemmas.k

```diff
@@ -330,6 +341,44 @@ module LEMMAS-HASKELL [symbolic, kore]
     rule BA:ByteArray   ==K #buf(32, DATA) => #buf(32, DATA) ==K              BA                                       [simplification, concrete(BA)]
     rule #buf(32, DATA) ==K BA:ByteArray   =>          DATA  ==Int #asInteger(BA) requires #sizeByteArray(BA) <=Int 32 [simplification, concrete(BA)]
 
+    rule #range( M [ N1 := BUF1:ByteArray ] [ N2 := BUF2:ByteArray ] [ N3 := BUF3:ByteArray ], L, K ) =>
+         #range( M                          [ N2 := BUF2           ] [ N3 := BUF3           ], L, K )
+      requires N1 +Int #sizeByteArray(BUF1) <=Int L [simplification]
+
+    rule #range ( M [ N := BUF:ByteArray ], N, K ) => #range( #range( M, N, K ) [ 0 := BUF ], 0, K ) 
+      requires N >Int 0 [simplification]
+
+    rule #range ( M, N, K ) => #range( #drop(N, M), 0, K ) 
+      requires N >Int 0 [simplification]
+
+    rule #range ( M [ N := BUF:ByteArray ], 0, K ) => M [ N := BUF ]
+      requires (#sizeByteArray( M ) ==Int K andBool #sizeByteArray( BUF ) <=Int K)
+        orBool (#sizeByteArray( M ) <=Int K andBool #sizeByteArray( BUF ) ==Int K)  [simplification]
+
+    rule M [ 0 := BUF:ByteArray ] => #padRightToWidth( #sizeByteArray( M ), BUF )
+      requires M ==K #padToWidth( #sizeByteArray( M ), .ByteArray ) 
+      andBool  #sizeByteArray( BUF ) <=Int #sizeByteArray( M )  [simplification]
+
+    rule #asWord( #padToWidth ( _, BUF ) ) => #asWord( BUF ) [simplification]
+
+    rule #range ( padRightBytes ( BUF:ByteArray, K, 0 ), 0, K ) => #padRightToWidth ( K, BUF ) [simplification]
```

Comment:
Should not have to mention `padRightBytes` in rules. Likely this is here because of the rule above which introduces it.

Reply:
This is gone now and `padRightBytes` is no longer introduced.

Reply:
Nice.